### PR TITLE
Cmseibel

### PR DESCRIPTION
**Pull request by CameronMSeibel**
Expands existing unit testing in services for increased coverage

### CODE REVIEW

File: src/test/java/com/revature/services/AddressServiceTest.java [hunk 1]
```diff
@@ -88,6 +88,24 @@ void testUpdate() {
...
+		users.add(this.dummyUser);
+		AddressRequest updateRequest = new AddressRequest(this.dummyAddress.getStreet(),
+				this.dummyAddress.getSecondary(), this.dummyAddress.getCity(), this.dummyAddress.getZip(),
+				this.dummyAddress.getState());
+		this.dummyAddress.setUsers(users);
+		int id = 404;
+		given(this.mockAddressRepo.findById(id)).willReturn(Optional.empty());
+		try {
+			this.aServ.update(updateRequest, id, this.dummyUser);
+			fail("Expected an exception to be thrown");
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/AddressServiceTest.java]: You should specify which exception type is expected to be thrown. In this case, it's `AddressNotFoundException`

File: src/test/java/com/revature/services/ProductServiceTest.java [hunk 2]
```diff
@@ -79,6 +80,18 @@ void testFindById() {
...
 		verify(this.mockProductRepo, times(1)).findById(id);
 	}
+	
+	@Test
+	void testFindById_Failure_ProductNotFound() {
+		int id = 404;
+		given(this.mockProductRepo.findById(id)).willReturn(Optional.empty());
+		try {
+			this.pServ.findById(id);
+			fail("Expected an exception to be throw");
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/ProductServiceTest.java]: You should specify which exception type is expected to be thrown. In this case, it's `ProductNotFoundException`

File: src/test/java/com/revature/services/ProductServiceTest.java [hunk 3]
```diff
@@ -110,6 +123,18 @@ void testDelete() {
...
 		verify(this.mockProductRepo, times(1)).deleteById(this.dummyProduct.getId());
 	}
+	
+	@Test
+	void testDelete_Failure_ProductNotFound() {
+		int id = 404;
+		given(this.mockProductRepo.findById(id)).willReturn(Optional.empty());
+		try {
+			this.pServ.delete(id);
+			fail("Expected an exception to be thrown");
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/ProductServiceTest.java]: You should specify which exception type is expected to be thrown. In this case, it's `ProductNotFoundException`

File: src/test/java/com/revature/services/PurchaseServiceTest.java [hunk 4]
```diff
@@ -93,6 +95,19 @@ void testFindByOwner() {
...
 	}
+	
+	@Test
+	void testFindByOwner_Failure_UserNotFound() {
+		int id = 2;
+		given(this.uServ.findById(id)).willReturn(Optional.empty());
+		
+		try {
+			this.purchaseServ.findByOwner(id);
+			fail("Expected exception to be thrown");
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/PurchaseServiceTest.java]: You should specify which exception type is expected to be thrown. In this case, it's `UserNotFoundException`

File: src/test/java/com/revature/services/ReviewServiceTest.java [hunk 5]
```diff
@@ -96,6 +120,23 @@ void testAdd_Failure_ProductNotFound() {
...
+		ReviewRequest request = new ReviewRequest(this.dummyProduct.getId(), 1, "Not working",
+				"It doesn't work as advertised. I am dissatisfied with this product.");
+		int productId = this.dummyProduct.getId();
+		given(this.pServ.findById(productId)).willReturn(Optional.of(this.dummyProduct));
+		given(this.rServ.findByProductId(this.dummyProduct.getId())).willReturn(Collections.singletonList(
+				new Review(2, 1, "A title", "A review of a thing.", null, null, this.dummyProduct, this.dummyUser)));
+		
+		try {
+			this.rServ.add(request, dummyUser);
+			fail("Expected an exception to be thrown");
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/ReviewServiceTest.java]: You should specify which exception type is expected to be thrown. In this case, it's `DuplicateReviewException`

File: src/test/java/com/revature/services/ReviewServiceTest.java [hunk 6]
```diff
@@ -185,6 +226,20 @@ void testFindById() {
...
 	}
 
+	@Test
+	void testFindById_Failure_ReviewIdNotFound() {
+		int id = 2;
+		given(this.mockReviewRepo.findById(id)).willReturn(Optional.empty());
+
+		try {
+			this.rServ.findById(id);
+			fail("Expected an exception to be thrown");
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/ReviewServiceTest.java]: You should specify which exception type is expected to be thrown. In this case, it's `ReviewNotFoundException`

File: src/test/java/com/revature/services/UserServiceTest.java [hunk 7]
```diff
@@ -52,6 +54,19 @@ void testFindByCredentials() {
...
+	@Test
+	void testFindByCredentials_Failure_UserNotFound() {
+		String email = "user@coolmail.edu";
+		String password = "Alligator3";
+		given(this.mockUserRepo.findByEmailAndPassword(email, password)).willReturn(Optional.empty());
+		try {
+			this.uServ.findByCredentials(email, password);
+			fail("Expected an exception to be thrown");
+		} catch (Exception e) {
+			assertEquals(UserNotFoundException.class, e.getClass());
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/UserServiceTest.java:67]: You should specify which exception type is expected to be thrown. In this case, it's `UserNotFoundException`

File: src/test/java/com/revature/services/UserServiceTest.java [hunk 8]
```diff
@@ -64,6 +79,18 @@ void testFindById() {
...
 		verify(this.mockUserRepo, times(1)).findById(id);
 	}
+	
+	@Test
+	void testFindById_Failure_UserNotFound() {
+		int id = 1337;
+		given(this.mockUserRepo.findById(id)).willReturn(Optional.empty());
+		try {
+			this.uServ.findById(id);
+			fail("Expected an exception to be thrown");
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/UserServiceTest.java]: You should specify which exception type is expected to be thrown. In this case, it's `UserNotFoundException`

File: src/test/java/com/revature/services/UserServiceTest.java [hunk 9]
```diff
@@ -87,5 +114,16 @@ void testFindByEmail() {
 		assertEquals(expected, actual);
 		verify(this.mockUserRepo, times(1)).findByEmail(email);
 	}
+	
+	@Test
+	void testFindByEmail_Failure_UserNotFound() {
+		String email = "user@coolmail.gov";
+		given(this.mockUserRepo.findByEmail(email)).willReturn(Optional.empty());
+		try {
+			this.uServ.findByEmail(email);
```
Comment (acarasimon96) [inline src/test/java/com/revature/services/UserServiceTest.java:123]: You forgot to add the following line after:
```java
fail("Expected UserNotFoundException to be thrown");
```

**Review by acarasimon96 — CHANGES_REQUESTED** [review]
Failure messages need tweaking, but this looks great to me otherwise

**Comment by CameronMSeibel** [comment]
Implemented changes to service fail messages, may want to hold off on merging this branch until the controller tests work again. Currently can't build with test failures, so I can't generate a coverage report

**Pull request closed without merging**